lib/vyasa_web/components/core_components.ex

```diff
@@ -1063,4 +1076,90 @@ defmodule VyasaWeb.CoreComponents do
     </div>
     """
   end
+
+  @doc """
+  Renders a sheet (modal sliding from the side/top).
+
+  Leverages existing modal-related JS functions (show_modal, hide_modal)
+  for consistency with other modal components.
+  """
+  attr :id, :string, required: true, doc: "Unique ID for the sheet."
+  attr :show, :boolean, default: false, doc: "Controls the visibility of the sheet."
+
+  attr :side, :atom,
+    values: [:top, :bottom, :left, :right],
+    default: :right,
+    doc: "The side from which the sheet slides in."
+
+  attr :on_cancel, JS,
+    default: %JS{},
+    doc: "JS commands to execute when the sheet is closed/cancelled."
+
+  attr :close_button, :boolean, default: true, doc: "Show a close button."
+  attr :trap_focus, :boolean, default: true, doc: "Trap focus within the sheet."
+
+  attr :esc_to_close, :boolean,
+    default: true,
+    doc: "Close the sheet when the Escape key is pressed."
+
+  attr :click_away_to_close, :boolean,
+    default: true,
+    doc: "Close the sheet when clicking outside."
+
+  slot :trigger, doc: "The element that triggers the sheet to open."
+  slot :content, required: true, doc: "The content of the sheet."
+
+  def sheet(assigns) do
```

Comment:
this is redundant, not being used and can be rm 

i'm currently just using modal wrapper directly